pydomcfg/domzgr/zgr.py

```diff
@@ -25,131 +25,121 @@ def __init__(self, ds_bathy: Dataset, jpk: int):
              number of computational levels
         """
 
-        self._bathy = ds_bathy
+        # Input arguments
+        self._bathy = ds_bathy.copy()
         self._jpk = jpk
 
-    # -------------------------------------------------------------------------
-    def _init_ds(self):
-        """
-        Initialise the xarray dataset with empty
-        ``z3{T,W}`` and ``e3{T,W}``
-
-        Returns
-        -------
-        ds: Dataset
-            A copy of the dataset used to initialise the class with new
-            coordinates ``z3{T,W}`` and ``e3{T,W}`` empty dataarrays
-
-        """
-        ds = self._bathy.copy()
-
-        var = ["z3", "e3"]
-        grd = ["T", "W"]
-
-        # Initialise a dataset with empty z3 and e3 dataarrays
-        da = xr.full_like(ds["Bathymetry"], None).expand_dims(z=range(self._jpk))
-        for v, g in product(var, grd):
-            ds[v + g] = da.copy()
-            ds = ds.set_coords(v + g)
-        return ds
```

Comment:
We don't need to initialize the dataset anymore. If you want to start Sco with a similar approach (i.e., start with loops, then remove them) you'd have to re-introduce this.